pycloudlib/lxd/instance.py

```diff
@@ -11,20 +12,31 @@ class LXDInstance(BaseInstance):
 
     _type = 'lxd'
 
-    def __init__(self, name):
+    def __init__(self, name, is_vm=False, key_pair=None):
         """Set up instance.
 
         Args:
             name: name of instance
+            is_vm: Specify if instance is a vm or not
+            key_pair: SSH key object
         """
-        super().__init__(key_pair=None)
+        super().__init__(key_pair=key_pair)
 
         self._name = name
+        self._is_vm = is_vm
 
     def __repr__(self):
         """Create string representation for class."""
         return 'LXDInstance(name={})'.format(self.name)
 
+    def _run_command(self, command, stdin):
+        """Run command in the instance."""
+        if self.key_pair:
+            return super()._run_command(command, stdin)
+
+        base_cmd = ['lxc', 'exec', self.name, '--']
```

Comment:
I think we should default to non-root user for lxc exec commands (as we do in ua-tools integration tests)
```suggestion
        base_cmd = ['lxc', 'exec', "--user", "1000", self.name, '--']
```